docs/content/guides/author-apps/dapr/how-to-dapr-building-block/index.md

```diff
@@ -0,0 +1,100 @@
+---
+type: docs
+title: "How-To: Add a Dapr building block"
+linkTitle: "Add a Dapr building block"
+description: "Learn how to add a Dapr building block to a Radius Application"
+weight: 300
+categories: "How-To"
+tags: ["Dapr"]
+---
+
+This how-to guide will provide an overview of how to:
+
+- Leverage a [Dapr building block](https://docs.dapr.io/developing-applications/building-blocks/) with your Radius Application
+
+## Prerequisites
+
+- [Supported Kubernetes cluster]({{< ref "/guides/operations/kubernetes/overview#supported-kubernetes-clusters" >}})
+- [rad CLI]({{< ref getting-started >}})
+- [Dapr CLI](https://docs.dapr.io/getting-started/install-dapr-cli/)
+- [Radius initialized with `rad init`]({{< ref howto-environment >}})
+- [Dapr initialized with `dapr init -k`](https://docs.dapr.io/getting-started/install-dapr-selfhost/)
+
+## Step 1: Start with a container with a Dapr sidecar
+
+Begin by creating a file named `app.bicep` with a defined Radius container and a Dapr sidecar, if you need a in-depth walkthrough see the ['How-To: Add a Dapr sidecar to a container' guide]({{< ref how-to-dapr-sidecar >}}):
+
+{{< rad file="../how-to-dapr-sidecar/snippets/app-sidecar.bicep" embed=true >}}
+
+## Step 2: Add a Dapr state store resource
+
+Now add the Dapr state store resource, which will allow Dapr's state management API to save, read and query key/value pairs for resources such as a Redis Cache:
+
+{{< rad file="./snippets/app-statestore.bicep" embed=true marker="//STATESTORE" >}}
+
+## Step 3: Add a connection from the container resource to the Dapr state store resource
+
+Edit your container resource to allow the Dapr state store to establish a connection:
```

Comment:
```suggestion
Update your container resource with a connection to the Dapr state store. This will inject important connection information (the component name) into the container's environment variables:
```